### DISCUSSION TITLE
Comparison to espalona

### DISCUSSION THREAD

**Discussion by tueboesen** [mir-group/nequip#168] [Q&A] [answered]
I was wondering whether you guys have any experience/comparisons of how espaloma compares to nequip?

My understanding these two neural networks basically do the same thing. They both take atomic positions and types as input and calculate the potential energy/force field as output. With the difference being that espaloma is designed to immitate the conventional force terms using bonds angles and torsions, while nequip accounts for having the right SE(3) symmetries through equivariance, but doesn't otherwise try to immitate any of the conventional force terms.
(For more info on espaloma, I found this presentation quite good https://www.youtube.com/watch?v=OC210nUuXHk)

I searched to see whether I could find anything about any such comparisons, but nothing came up, which is why I decided to ask the question.

Note that I asked the reverse question on the espaloma github page to try and get both perspectives on this:
https://github.com/choderalab/espaloma/issues/109

**Comment by simonbatzner** [comment, marked as answer]
Hi @tueboesen, yes, we've come across the espaloma work. Both NequIP and espaloma are after the same high-level goal of predicting E/F from positions, but keep in mind that there are many ways of achieving this goal (for a list of other potentials, see the NequIP paper). The ideas between espaloma and NequIP are actually quite different I would say, in particular as you say, NequIP builds an E(3)-equivariant neural network that does not build in any explicit conventional terms while Espaloma imitates these terms. 

Espaloma war proposed quite a while after NequIP and while their work shows tons of interesting data, I don't believe a comparison to more conventional ML force-fields exists yet. We benchmark all our results on the revMD17 data set which is widely used in the field. It would certainly be interesting to see how well Espaloma would do on that benchmark. 

As hypothesized in the video you linked it is likely that the Espaloma approach is faster but less accurate than SOTA ML Potentials like NequIP. I believe that is a valid assumption to make.

**Reply by tueboesen** [reply]
Thank you for the prompt reply!

The thing that surprises me about espaloma is that they build these explicit terms to emulate conventional force fields, yet they also seem to be able to predict both QM force fields and peptides (which I believe should be treated as if they are in soluble water).

I would imagine that Nequip should be better at both those two cases, since it makes no such assumptions.

Finally I'm a bit confused by what you mean when you say Espaloma should be faster than NequIP? they are both ML approaches, so their speed should be comparable and really only depend on the neural network size, or do you mean that Espaloma might be faster to train/require less training due to the explicit way it is built?

**Reply by simonbatzner** [reply]
Yes, NequIP will likely be more accurate given it's flexibility. Regarding speed: my understanding of espaloma is that it only has to generate the MM parameters **once at the beginning** and then these parameters are fixed and you'd only evaluate the much, much cheaper functional form of the classical FF (maybe @jchodera can confirm this understanding).
